### DISCUSSION TITLE
[Extend with JS] TypeError: Object has no member 'get' at <eval>:10:36(52)

### DISCUSSION THREAD

**Discussion by bijsven** [pocketbase/pocketbase#4055] [Q&A] [answered]
The code starts at line 7. So, UserID is line 10.
```js
routerAdd("POST", "/subjects/join", (c) => {
    const SubjectID = ($apis.requestInfo(c).data).subject;
    const UserID = $app.dao().findAuthRecordByToken(c.request().header.get("token"), $app.settings().recordAuthToken.secret).id

    var Subject = $app.dao().findRecordById("subjects", SubjectID);
    if(!Subject) {throw new Error("Subject does not exist.")}

    var SubjectOwner = Subject.get("owner");
    SubjectOwner = SubjectOwner.get("id")

    if(UserID == SubjectOwner) {
        return c.json(200, { "status": "ERROR" });
    }

    var SubjectUsers = Subject.get("users");
    SubjectUsers.push(UserID)

    Subject.set("users", SubjectUsers);
    $app.dao().saveRecord(Subject);
   
    return c.json(200, { "status": "OK" })
});
```

I couldn't seem to find the error, if i'm following the documentation, its says this:

```js
const token = c.request().header.get("Some-Header")
```

So i don't know what I am doing wrong...

**Comment by ganigeorgiev** [comment, marked as answer]
_Note that because of the hooks function serialization the line count practically starts from the beginining of the callback function._

In your case, the error seems to be `SubjectOwner = SubjectOwner.get("id")`.
This is invalid because the previous assignment `Subject.get("owner")` returns a string (which is the id of the relation).

**Reply by ganigeorgiev** [reply]
_Also note that `$app.dao().findRecordById("subjects", SubjectID);` will throw an error if no record matching the `SubjectID` is found so the line `if(!Subject) {throw new Error("Subject does not exist.")}` is unnecessary and will never get triggered._

**Reply by bijsven** [reply]
Thanks! Amazing work on Pocketbase btw!
